File: rezmate_backend/YemenBooking.Core/Enums/BookingStatus.cs
namespace YemenBooking.Core.Enums;

/// <summary>
/// تعداد حالات الحجز
/// Booking Status enumeration
/// </summary>
public enum BookingStatus
{
    /// <summary>
    /// مؤكد
    /// Confirmed
    /// </summary>
    Confirmed,
    
    /// <summary>
    /// معلق
    /// Pending
    /// </summary>
    Pending,
    
    /// <summary>
    /// ملغى
    /// Cancelled
    /// </summary>
    Cancelled,
    
    /// <summary>
    /// مكتمل
    /// Completed
    /// </summary>
    Completed,
    
    /// <summary>
    /// تسجيل الوصول
    /// Checked-in
    /// </summary>
    CheckedIn
} 

File: hggzk_app/lib/core/enums/booking_status.dart
enum BookingStatus {
  confirmed,
  pending,
  cancelled,
  completed,
  checkedIn,
}

extension BookingStatusExtension on BookingStatus {
  String get displayName {
    switch (this) {
      case BookingStatus.confirmed:
        return 'مؤكد';
      case BookingStatus.pending:
        return 'في انتظار التأكيد';
      case BookingStatus.cancelled:
        return 'ملغى';
      case BookingStatus.completed:
        return 'مكتمل';
      case BookingStatus.checkedIn:
        return 'تم تسجيل الوصول';
    }
  }

  String get displayNameEn {
    switch (this) {
      case BookingStatus.confirmed:
        return 'Confirmed';
      case BookingStatus.pending:
        return 'Pending';
      case BookingStatus.cancelled:
        return 'Cancelled';
      case BookingStatus.completed:
        return 'Completed';
      case BookingStatus.checkedIn:
        return 'Checked In';
    }
  }
}